### ISSUE TITLE
AIP 44: Diversify the AST treasury with UMA Range tokens

### ISSUE THREAD

**Issue by agriimony** [airswap/AIPs#44]
Author - Greg P

## Summary

Proposal for AirSwap to mint and sell a set of “range tokens” in collaboration with UMA to diversify treasury and raise some stablecoin for operational reasons.

## Rationale

There is a lot of research that DAO’s should have a diversified treasury. At the moment, all we hold is AST. Crypto markets are cyclical and diversifying to hold some stablecoin helps protect the financial future of AirSwap.

Some benefits:

* Access funds and diversify treasury without directly selling $AST tokens
* Borrow funds without liquidation using AST as collateral
* Define lock up terms for investors in smart contract
* This allows AirSwap to effectively leverage (take on debt) against its treasury now, with the hope of paying back that debt at a higher AST price than today in order to ensure it has the necessary stable coin funds for various operational expenses it may want to pay out
* AirSwap retains the upside on the $AST up to the max of the range.

## Specification

Range Tokens are minted with a specified upper and lower bounds which represent a claim on a number of AST tokens.

We would need to consider

* Length of contract
* Range token price
* Low price bound
* Upper price bound
* Amount of AST collateral* (amount stablecoin to net)

Perhaps 10% of the unlocked budget = 10M AST. We will return the USD stablecoin to each pool (marketing / growth / dev) respectively?

Copyright — All proposals are public domain via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).

edit - adding a couple of useful links
Reference:

Treasury Diversification With Range Tokens
https://medium.com/uma-project/treasury-diversification-with-range-tokens-145d4b12614e

Shapeshift have a proposal where this same idea is being discussed
https://forum.shapeshift.com/t/mint-and-sell-range-tokens-for-treasury-diversification/136/18

**Comment by agriimony** [comment]
agrimony
Jul 31

Adding an example here:

AirSwap could sell a 1.5M USD worth of AST collateralized RTs at $0.15 with a lower bound of $0.05 and an upper bound of $0.5 which matures in 1 year. We would lock 30M AST from the treasury in the range token contract (assuming current price is $0.15) to mint 10M RTs and sell them at $0.15 per RT.

After a year, if the AST price drops to $0.05, range token holders can exercise the option to claim 3 AST per RT. RT holders will be able to claim up to a maximum of 3 AST per RT even if the price drops below $0.05.

On the other hand, if the price increases to $0.50, RT holders can exercise the option to claim 0.3 AST per RT. RT holders will always be able to claim a minimum of 0.3 AST per RT, thus if the price moves above $0.5, they will profit from their investment.

----

agrimony
Jul 31

We can also offer a discount (~10%?) on the current price, thus allowing range token holders to earn some profit even if the price does not move out of the range.

----

Inalittlewhile
Jul 31

Inalittlewhile here as a SuperUMAn ambassador. Thanks for doing the leg work to draft and put this proposal up for adoption Greg.

Don’t want to steer the conversation as much as offer full support from team UMA. Both from the SuperUMAn DAO (uDAO), and the UMA core team. Feel free to shoot with any questions. We’ll be around to answer.

Inalittlewhile - Team UMA

----

clayton
Aug 1

Hi @greg Thank you for submitting this proposal. I’m the Community Lead at UMA and we would be excited to work together on getting Airswap’s treasury diversified.

We’d love to have you and your community over to our DAO stage some time this week to meet-and-greet everyone, get questions answered, and so forth. You can coordinate with inalittlewhile to move that forward for sure.

@agrimony Yes, you’d want to offer a discount on the sale, that’s the part of the product that results in yield. E.g., if you sold them for .90, then an in-range expiry would realize a 11.1% APY for buyers. That’s what Airswap is paying to borrow the stablecoins.

----

VladislavW

Nice idea! As always! My thoughts about the variables:

    Length of contract - One year should be a good term
    Range token price - 10% discount with 11.1% minimum APY for stablecoin deposit is a good competitive option on the market
    Low price bound and Upper price bound - $0.05 - $0.5 is a perfect range because AST price never dropped much lower than $0.1 but i believe will cost much more than $0.5 next year what will be a nice investment
    Amount of AST collateral* (amount stablecoin to net) - 10M AST which would be $1.5M is a nice budged for one year to develop and promote the project. But maybe we need to increase it to $2-2.5M to have a buffer, anyway it would be charged back if not used.

----

clayton
Aug 2

@VladislavW I think those considerations all sound great. To be honest, I think getting something off the ground will be the very best starting point. For example, you can go ahead and do this 1-year contract, with those bounds, and start selling them.

You don’t have to sell all of them - Nor stay committed to those designs. E.g., you could realize you want to do 2-year product, and then allocate additional tokens or even withdraw from the first product (any you haven’t already sold.)

Getting something happening is the best way to realize if terms should be tweaked. Your proposal seems like a very sane starting point @VladislavW.

I think the way forward is:

    Get the community’s approval to put your and UMA’s attention on this, which perhaps this post is sufficient for?
    Put the “stage one” proposal out and ask it to be passed

How does that sound?

-----

agrimony
Aug 5

Could work as well.

$2M would be 13.3M AST, so would need to lock away 40M AST.

Just a note - from AIP 10, we have a total of 98.3M AST available which is meant to pay for things like devt / growth and marketing.

The AST locked away to mint the range tokens would have to be spread out equally among those budgets. The USD gained from selling the range tokens would be equally spread out among the groups, and can be used for marketing campaigns etc where USD would come in handy.

In other words, we would need to lock up approximately 40% of the available AST (40/98M), and in return generate $2M to be used for the same purposes.
